qml/pages/ChatPage.qml

```diff
@@ -543,6 +577,19 @@ Page {
 
         PullDownMenu {
             visible: chatInformation.id !== chatPage.myUserId && !stickerPickerLoader.active && !messageOverlayLoader.active
+            MenuItem {
+                id: closeSecretChatMenuItem
+                visible: chatPage.isSecretChat && chatPage.secretChatDetails.state["@type"] !== "secretChatStateClosed"
+                onClicked: {
+                    var remorse = Remorse.popupAction(appWindow, qsTr("Closing chat"), (function(secretChatId) {
+                        return function() {
+                            tdLibWrapper.closeSecretChat(secretChatId);
+                        };
+                    }(chatPage.secretChatDetails.id)))
```

Comment:
Is there a reason for this being done in such a complicated manner? I'm not even sure how this is supposed to work. Remorse timers usually look much simpler than that.

Reply:
Yes, if the page is closed, the variable is still available because it is "trapped" in the function scope.

Reply:
Wouldn't something like this do it?
```JavaScript
    var chatId = chatPage.secretChatDetails.id
    Remorse.popupAction(appWindow, qsTr("Closing chat"), function() {
        tdLibWrapper.closeSecretChat(chatId)
    })
```

Reply:
In this case I simply copied the logic over from the previous example by @jgibbon - if we change that, I'd like to do it for all of the similar examples...

Reply:
I think (don't quote me on that,yet) I initially had the simpler version, because that would make sense and would work in "pure JS" or with primitive types, but IIRC it didn't.

If I get some time tonight (started self isolation today, many things to do/plan) I'll check again. 
I propose to leave it here as it is and if I confirm it's unnecessary, I'll make a separate PR (perhaps an issue).

Reply:
Sure, as long as it doesn't get forgotten.

It's just quite typical that something is initially being done in an overly complicated manner for some more or less valid reason, then that reason completely goes away but the complexity continues to be dragged along, contaminating other things along the way.

Life is a (hopeless) fight against entropy :)

Reply:
I have tested this now. 
And while it really didn't work before, the error is something different: 
For some (aka. apparently no) reason I called it with `appWindow` as a parent, so it just dies regardless of how complicated the rest is, so even that workaround is useless.
When using chatPage as it is intended, the simple solution works. Thanks for noticing!

I've created #206 to get rid of this.